Examples/MAX78000/Flash_CLI/main.c

```diff
@@ -122,6 +135,53 @@ void vCmdLineTask_cb(mxc_uart_req_t *req, int error)
     portYIELD_FROM_ISR(xHigherPriorityTaskWoken);
 }
 
+/* =| checkLeadingSpaces |======================================
+ *
+ * Function to check for the leading spaces in the 
+ *  input command.
+ *
+ * If any Leading spaces are present, the function would remove 
+ *  those spaces and returns a valid Null terminated command 
+ *  in the same input buffer.
+ * 
+ * =============================================================
+ */
+void checkLeadingSpaces(char *buffer, unsigned int *index)
+{
+    unsigned int leadingZerosCount = 0;
+    for (leadingZerosCount = 0; leadingZerosCount < (*index); leadingZerosCount++) {
+        if (buffer[leadingZerosCount] != SPACE_BAR)
+            break;
+    }
+    if (leadingZerosCount > 0) {
+        memcpy(buffer, buffer + leadingZerosCount, (*index) - leadingZerosCount);
```

Comment:
memcpy is not guaranteed to handle overlapping src and dest buffers.  This probably works with our default standard libraries, but other versions may not.  memmove is a better choice.

Reply:
Sure. Updated the code with memmove() to avoid overlapping buffers and tested it. 
Working fine now and I pushed it.